fix(ui): match table styles with PF5

ciiay opened the pull request:
## Description

Match table styles with PF5.

## Which issue(s) does this PR fix

- Fixes [RHIDP-1707](https://issues.redhat.com/browse/RHIDP-1707): Add MUI theme config so that Tables look similar to PF5

## PR acceptance criteria

Please make sure that the following steps are complete:

- [ ] GitHub Actions are completed and successful
- [ ] Unit Tests are updated and passing
- [ ] E2E Tests are updated and passing
- [ ] Documentation is updated if necessary (requirement for new features)
- [x] Add a screenshot if the change is UX/UI related

## Screen recording(updated on 4.5)

https://github.com/janus-idp/backstage-showcase/assets/26255262/5bfd8597-2bcb-44c1-9691-d9c23dcd425a

gashcrumb reviewed (approved):
Tried this locally, this is a nice improvement!

invincibleJai commented:
Thanks @ciiay , for theme PRs it's good to put the screenshot/gif for before and after so that it's easier for reviewer 

cc @jerolimov

ShiranHi commented:
@ciiay it looks really good, I am so happy to see this! 🙂
I have two questions:

1. What are the arrows on top?
<img width="1052" alt="image-1" src="https://github.com/janus-idp/backstage-showcase/assets/77277517/0bcf8f1d-1c10-4be9-888a-89963a69844c">

2. I'm not sure if we can tweak that, but the way things are sorted seems kinda weird to me. So, like, in this table below, it's sorted by "Description," and that text is all bolded. But the "Name" text still looks darker compared to the rest, like "System." Think we could adjust that to match more with the [PF component](https://www.patternfly.org/components/table/#sortable---custom-control)?
<img width="1039" alt="image-2" src="https://github.com/janus-idp/backstage-showcase/assets/77277517/a84fb961-5247-474c-9c79-186f32674b3d">

ciiay commented:
Hi @ShiranHi , thanks for the review. I have updated the table column title behavior to match PF5 styles. 
Key changes:
1. Default: bold and #151515(black)
2. Sorting: both title and sorting arrow in blue color
3. Matched dark mode color too
I also updated the screen recording. Let me know if it looks good now.

>What are the arrows on top?
These are paginate arrows which are from upstream component. It's not a style issue as we can't remove it by overriding MUI styles.

ShiranHi commented:
@ciiay looks great, and thank you for the clarification

invincibleJai commented:
/approve
